hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/PutObjectOptions.java

```diff
@@ -39,15 +49,36 @@ public final class PutObjectOptions {
   /**
    * Constructor.
    * @param storageClass Storage class, if not null.
+   * @param conditionalPutEnabled Is this a conditional Put?
    * @param headers Headers; may be null.
    */
   public PutObjectOptions(
+      final boolean keepMarkers,
```

Comment:
the keepmarkers option is something I've cut from trunk. It's come in here during rebase merge work.

proposed;
* you move the new argument to after the `headers` entry
* remove the keepmarkers argument/field and all uses of it in this class.